### PR TITLE
🚀 Add robust schema validation for handling null and empty fields

### PR DESCRIPTION
**Pull request by mohamedmamdouh22**
This feature introduces enhanced schema validation capabilities to ensure that null and empty fields are properly handled across various schemas. Key improvements include:

1. New Validation Logic: Implemented a function to raise SchemaValidationError when all fields in a data object are null or empty handling nested null fields.
2. Create a Custom Test For This feature + e2e testing. 
3. Support for Complex Schemas: Added validation for nested objects, arrays, and various data types, including strings, booleans, datetimes, and custom types like currency and phone_number.

### CODE REVIEW

**Review by asim-shrestha — APPROVED** [review]

File: harambe/parser/parser.py [hunk 1]
```diff
@@ -159,6 +165,25 @@ def _get_type(self, field: str) -> Type[Any]:
...
+
+        def is_empty(value: Any) -> bool:
+            if value is None:
+                return True
+            if isinstance(value, (str, list, dict)):
+                if isinstance(value, dict):
+                    return all(is_empty(v) for v in value.values())
+                if isinstance(value, list):
+                    return all(is_empty(v) for v in value)
+                return len(value) == 0
```
Comment (awtkns) [inline harambe/parser/parser.py:182]: some needless nesting here

```suggestion
             if isinstance(value, dict):
                 return all(is_empty(v) for v in value.values())
             if isinstance(value, list):
                 return all(is_empty(v) for v in value)
             if isinstance(value, str)
              return bool(value)
```

File: test/parser/test_parser.py [hunk 2]
```diff
@@ -303,6 +291,26 @@ def test_pydantic_schema_validator_success(
...
+            schemas.documents_schema,
+            {
+                "documents": []  # ❌ Do not allow objects with all null arrays
+            },
+        ),
+        (
+            schemas.document_schema,
+            {
+                "title": "",  # ❌ Do not allow objects with all empty strings
+                "document_url": "",
```
Comment (awtkns) [inline test/parser/test_parser.py:311]: might be good to add a test for an object with a list of empty dicts eg: {list: [{}, {a: None, b: {}, c: []]}

**Review by awtkns — APPROVED** [review]

File: harambe/parser/parser.py [hunk 3]
```diff
@@ -159,6 +165,25 @@ def _get_type(self, field: str) -> Type[Any]:
...
+
+        def is_empty(value: Any) -> bool:
+            if value is None:
+                return True
+            if isinstance(value, dict):
+                return all(is_empty(v) for v in value.values())
+            if isinstance(value, list):
+                return all(is_empty(v) for v in value)
+            if isinstance(value, str):
+                return not bool(value.strip())
```
Comment (awtkns) [inline harambe/parser/parser.py:182]:
```suggestion
                return not value.strip()
```
